src/app/GitUI/Editor/Diff/DiffHighlightService.cs

```diff
@@ -126,115 +127,137 @@ public override void AddTextHighlighting(IDocument document)
-    /// <param name="line">The line number to start with, updated with the last line processed.</param>
-    /// <param name="found">Ref updated if any removed lines were found.</param>
-    /// <returns>List with the segments of removed lines.</returns>
-    protected virtual List<ISegment> GetRemovedLines(IDocument document, ref int line, ref bool found)
-        => [];
+        text = sb.ToString();
+    }
 
     /// <summary>
-    /// Highlight the directly following lines.
-    /// Overridden in the HighlightServices where GE coloring is used (AddTextHighlighting() for Patch and CombinedDiff).
+    /// Highlight lines that are added, removed and header lines.
+    /// This is an alternative configuration to use the Git diff coloring (that has more features).
     /// </summary>
-    /// <param name="document">The document to analyze.</param>
-    /// <param name="line">The line number to start with.</param>
-    /// <param name="lineSegment">The segment for the starting line.</param>
-    /// <returns>The last line number processed.</returns>
-    protected virtual int TryHighlightAddedAndDeletedLines(IDocument document, int line, LineSegment lineSegment)
-        => line;
-
-    protected void ProcessLineSegment(IDocument document, ref int line,
-        LineSegment lineSegment, string prefixStr, Color color, bool invertMatch = false)
+    private void HighlightAddedAndDeletedLines(IDocument document)
     {
-        if (!DoesLineStartWith(document, lineSegment.Offset, prefixStr, invertMatch))
+        foreach (ISegment segment in GetAllLines(DiffLineType.Minus))
         {
-            return;
+            document.MarkerStrategy.AddMarker(CreateTextMarker(segment.Offset, segment.Length, _removedBackColor));
         }
 
-        LineSegment endLine = document.GetLineSegment(line);
+        foreach (ISegment segment in GetAllLines(DiffLineType.Plus))
+        {
+            document.MarkerStrategy.AddMarker(CreateTextMarker(segment.Offset, segment.Length, _addedBackColor));
+        }
 
-        for (;
-            line < document.TotalNumberOfLines
-            && DoesLineStartWith(document, endLine.Offset, prefixStr, invertMatch);
-            line++)
+        foreach (ISegment segment in GetAllLines(DiffLineType.Header))
         {
-            endLine = document.GetLineSegment(line);
+            document.MarkerStrategy.AddMarker(CreateTextMarker(segment.Offset, segment.Length, AppColor.DiffSection.GetThemeColor()));
         }
+    }
 
-        line = Math.Max(0, line - 2);
-        endLine = document.GetLineSegment(line);
+    /// <summary>
+    ///  Matches related removed and added lines in a consecutive block of a patch document and marks identical parts dimmed.
+    /// </summary>
+    private void MarkInlineDifferences(IDocument document)
+    {
+        int index = 0;
+        List<DiffLineInfo> diffLines = [.. _diffLinesInfo.DiffLines.Values.OrderBy(i => i.LineNumInDiff)];
```

Comment:
What does this syntax mean? `[.. orderedEnumerable]` in this single usage is a weird expression for `orderedEnumerable.ToArray()`. This ".." is named "spread element".
I prefer the explicit syntax although the Roslyn analyzer advertises the new syntax - not requesting a change.
```suggestion
        DiffLineInfo[] diffLines = _diffLinesInfo.DiffLines.Values.OrderBy(l => l.LineNumInDiff).ToArray();
```
but at least
```suggestion
        DiffLineInfo[] diffLines = [.. _diffLinesInfo.DiffLines.Values.OrderBy(l => l.LineNumInDiff)];
```

Reply:
spread operator, all from start to end. 
I do like the syntax, but as you want to change to an array I will change

Reply:
I will get used to that syntax. It can be used for array, too.
I.e. `ToArray()` is not necessary: My second suggestion in the initial comment works.

Reply:
applied the second suggestion